src/ffi/dial_ffi.rs

```diff
@@ -114,6 +116,7 @@ fn dial_with_cred(
 /// * `c_type` a C-style string representing the type of robot's secret you want to use, set to NULL if you don't need authentication
 /// * `c_payload` a C-style string that is the robot's secret, set to NULL if you don't need authentication
 /// * `c_allow_insecure` a bool, set to true when allowing insecure connection to your robot
+/// * `c_timeout` a float, set how long we should try dial before timing out
```

Comment:
[minor] 

```suggestion
/// * `c_timeout` a float, set how long we should try to dial before timing out
```

Reply:
Would mocking the implementation of `dial_with_cred` to be a time::sleep and making sure it times out a good test?

Reply:
Wait, or would the test be in Python?

Reply:
yeah that totally works, but before that I was wondering how straightforward it would be to call this function in a normal rust unit test context. if we need to build and call in another language then I would say it's not worth it.

Reply:
So, calling the extern func from within Rust is totally doable. However, in order to call it we have to initialize a new Rust runtime (see `init_rust_runtime` and the `rt_ptr` arg)

Tokio test does not seem to like this

```thread 'test_dial_ffi' panicked at 'Cannot start a runtime from within a runtime. This happens because a function (like `block_on`) attempted to block the current thread while the thread is being used to drive asynchronous tasks.'```

Reply:
Maybe this is now at the point where writing this particular test is too big of a hurdle for now ?

Reply:
yeah I think so, thank you for trying!